drivers/cadence/uart/aarch64/cdns_console.S

```diff
@@ -111,13 +111,13 @@ func console_cdns_core_putc
 1:
 	/* Check if the transmit FIFO is full */
 	ldr	w2, [x1, #R_UART_SR]
-	tbnz	w2, #UART_SR_INTR_TFUL_BIT, 1b
+	tbz	w2, #UART_SR_INTR_TEMPTY_BIT, 1b
```

Comment:
Doesn't this mean that you are flushing the FIFO on every putchar, which will slow down the overall UART throughput? Wouldn't it better to implement the empty console_core_flush function and ensure this is called prior to reset.

In any case, the comment in #112 is now inaccurate

Reply:
@danh-arm , 
Thanks for your comment. you are correct, its better to implement flush instead of modifying this. Will look into it as a separate patch outside this series. I will send next version with out this change.

Reply:
@danh-arm 
Also, removed unnecessary merge commit.